src/EditorFeatures/Core/Diagnostics/AbstractPushOrPullDiagnosticsTaggerProvider.SingleDiagnosticKindPullTaggerProvider.cs

```diff
@@ -109,6 +109,17 @@ protected sealed override ITaggerEventSource CreateEventSource(ITextView? textVi
 
             try
             {
+                // If this is not the tagger for compiler-syntax, then suppress diagnostics until the project has fully
+                // loaded.  This prevents the user from seeing spurious diagnostics while the project is in the process
+                // of loading.  We do keep compiler-syntax as that's based purely on the parse tree, and doesn't need
+                // correct project info to get reasonable results.
+                if (_diagnosticKind != DiagnosticKind.CompilerSyntax)
+                {
+                    var hasSuccessfullyLoaded = await document.Project.HasSuccessfullyLoadedAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
This code should be in the DiagnosticAnalyzerService GetDiagnosticsForXXX APIs so that all pull diagnostics clients get the benefit. This change will avoid the spurious squiggles in the editor while the project is still loading but you will still see these errors in the error list, we want to avoid both.

Reply:
yup.  totally agree.  this is a temp fix to just deal with teh squiggles.  The error list will currently be ok as that is still on push-diags for all scenarios (but would become pull once we do our lsp A/B test).

Reply:
can we take this while you work on a fix for the lower levels?

Reply:
Yes, that sounds reasonable

Reply:
Have you measured if there is any perf impact here? I see the code flows down to https://sourceroslyn.io/#Microsoft.CodeAnalysis.Workspaces/Workspace/Solution/SolutionState.CompilationTracker.cs,1023, which has a slow path that builds compilation info. I am wondering if that is the reason why the push diagnostics path also doesn't have this check for computing active/open file diagnostics, but only for entire project diagnostics.

Reply:
yeah, i'm going to wait on this to fully understand this bit.  will have to discuss with @jasonmalinowski .  IT's not clear why this needs the compilation, as opposed to just ensuring that the last design time build completed successfully.

Reply:
Yeah, looking into this, this does cause an unfortunate building of the compilations in the host process (as opposed to in oop where the diagnostics are computed).  I believe all we care about is if this project and its dependencies "have all information".  So i'm just going to compute off of that.  

Technically, it's a slightly different result than HasSuccessfullyLoadedAsync.  The latter not only needs all projects to "have all information", it also requires that skeletons be built.  The latter only matters though for cross language scenarios.  And we should be able to make a reasonable guess that if we loaded all info properly, it's likely skeletons will be successful at building.  

So it's worth doing for the normal case of single-language, or skeletons building fine, as it gets rid of noisy errors.  Worse case is that a cross-language file with skeletons still has some errant errors.  But they would already be bad today, so it's no worse.